### PR TITLE
Bug fix/rlp 838/fixing tests

### PR DESCRIPTION
**Pull request by JONAF2103**
This PR aims to fix all the broken tests for the refactor made on https://github.com/rsksmart/lumino/pull/119, it's based on this PR https://github.com/rsksmart/lumino/pull/142

* Fixing tests
* Fixing some bugs found by running the tests
* Fixing the test setup (this fixes a lot of broken tests)

This affects the tests

Tests checked with develop [baseline](https://github.com/anarancio/lumino-docs/blob/master/unit-testing/results/develop/2020-10-06/test_results.txt) and everything seems to be ok, here the [results](https://github.com/rsksmart/lumino/files/5438975/RLP-838.tests.log)

### CODE REVIEW

File: raiden/tests/integration/long_running/test_integration_events.py [hunk 1]
```diff
@@ -228,10 +228,6 @@ def run_test_channel_deposit(raiden_chain, deposit, retry_timeout, token_address
         registry_address, token_address, app0.raiden.address, app1.raiden.address, deposit
     )
 
-    wait_both_channel_deposit(app0, app1, registry_address, token_address, deposit, retry_timeout)
```
Comment (marcosmartinez7) [inline raiden/tests/integration/long_running/test_integration_events.py:231]: Could you explain why you removed this?

File: raiden/tests/integration/network/transport/test_matrix_transport.py [hunk 2]
```diff
@@ -85,7 +87,9 @@ def mock_receive_message(klass, message):  # pylint: disable=unused-argument
         private_rooms=private_rooms,
     )
 
-    transport = MatrixTransport(config)
+    address = to_checksum_address(privatekey_to_address(private_keys[0]))
+
+    transport = MatrixTransport(address=address, config=config)
```
Comment (marcosmartinez7) [inline raiden/tests/integration/network/transport/test_matrix_transport.py:92]: nice catch here :)

File: raiden/tests/integration/network/transport/test_matrix_transport.py [hunk 3]
```diff
@@ -151,7 +155,7 @@ def ping_pong_message_success(transport0, transport1):
     return all_messages_received
 
 
-def is_reachable(transport: MatrixTransport, address: Address) -> bool:
+def is_reaceqhable(transport: MatrixTransport, address: Address) -> bool:
```
Comment (marcosmartinez7) [inline raiden/tests/integration/network/transport/test_matrix_transport.py]: typo here, be careful cause this function is used on: 

- test_matrix_invite_private_room_unhappy_case2
- test_matrix_invite_private_room_unhappy_case3
- test_matrix_user-roaming

File: raiden/tests/integration/network/transport/test_matrix_transport.py [hunk 4]
```diff
@@ -944,8 +957,6 @@ def test_matrix_user_roaming(matrix_transports):
 
     transport0.start(raiden_service0, message_handler0, "")
 
-    assert is_reachable(transport1, raiden_service0.address)
```
Comment (marcosmartinez7) [inline raiden/tests/integration/network/transport/test_matrix_transport.py:947]: why this was removed?

File: raiden/transfer/views.py [hunk 5]
```diff
@@ -517,7 +517,10 @@ def get_transfer_role(chain_state: ChainState, node_address: AddressHex, secreth
     in a transfer. If a transfer task is not found for the secrethash then the
     function returns None
     """
-    task = chain_state.payment_mapping[node_address].secrethashes_to_task.get(secrethash)
+    task = None
```
Comment (marcosmartinez7) [inline raiden/transfer/views.py:520]: Some comments here:

1. why this was needed? 
2. its ok to protect ourselves of an exception but this scenario cant happen, if happens, we have a major issue, so i prefeer to keep the code as it was
3. i think that if a PR is for solving tests and we modify something of the core of lumino, that must be another pr (with, probably, another round of manual and unit tests)

**Review by marcosmartinez7 — CHANGES_REQUESTED** [review]
Great job in general

Some minor changes with naming :)

File: raiden/transfer/views.py (hunk 5 above)
Comment (JONAF2103) [reply raiden/transfer/views.py:520]: This was because you have chain_state.payment_mapping as a dict empty sometimes and accessing a dict without that key blows the code, the background of why this dict is empty some times i don't have it, but seems to me that when you delete the task you use del and del deletes the key also so you loose the access to the object with the secret hashes too. The tests were failing before this so that's why i added that.

File: raiden/tests/integration/network/transport/test_matrix_transport.py (hunk 4 above)
Comment (JONAF2103) [reply raiden/tests/integration/network/transport/test_matrix_transport.py:947]: Because of the change that we made on the user presence, now is reachable it returns always the same because of that so is not a valid call.

File: raiden/tests/integration/network/transport/test_matrix_transport.py (hunk 3 above)
Comment (JONAF2103) [reply raiden/tests/integration/network/transport/test_matrix_transport.py]: Yes good catch, i don't know how the complete set of tests pass with this :S

Comment (JONAF2103) [reply raiden/tests/integration/network/transport/test_matrix_transport.py]: Make sense, i removed all the usages so it's ok. I fixed this typo

File: raiden/tests/integration/long_running/test_integration_events.py (hunk 1 above)
Comment (JONAF2103) [reply raiden/tests/integration/long_running/test_integration_events.py:231]: Actually i just moved it to the end, since we were making the deposit for the first app and the waiting but the waiting method waits for both deposits so it was failing for timeout, so now we make both deposits and then we wait, so now it make sense, before this it was working on develop since the key was only the channel id and not the address, but it was wrong and now we can see that, so to fix it i just moved that to the end to wait for both apps to make the deposits

File: raiden/transfer/views.py (hunk 5 above)
Comment (marcosmartinez7) [reply raiden/transfer/views.py:520]: hmm but that situation of having the empty dict and accesing it, is related with some test? 

and if the answer is yes, why wasnt failing previously?

Comment (JONAF2103) [reply raiden/transfer/views.py:520]: Because before that you didn't access the dict with the node address, is like a nesting problem, before this you had something like this:
```python
chain_state.secrethashes_to_task.get(secrethash)
```
where secrethash could be or not in the dict and that was ok, since it never will fail, because if the hash wasn't there the if will check it, but now you have:
```python
chain_state.payment_mapping[node_address].secrethashes_to_task.get(secrethash)
```
where chain_state.payment_mapping can be = {} so the first "get" chain_state.payment_mapping[node_address] will be None and None can't be accessed directly so the rest of the call .secrethashes_to_task.get(secrethash) will fail for those cases.

Comment (JONAF2103) [reply raiden/transfer/views.py:520]: Maybe a better solution to this problem could be to make sure that when you create the dict chain_state.payment_mapping you add the addresses to the dict there, but i think that you don't have the addresses there so you need to make this check here.

**Review by marcosmartinez7 — APPROVED** [review]
